libtransmission/verify.cc

```diff
@@ -30,94 +23,22 @@ auto constexpr SleepPerSecondDuringVerify = 100ms;
 }
 } // namespace
 
-void tr_verify_worker::verify_torrent(Mediator& verify_mediator, std::atomic<bool> const& abort_flag)
+void tr_verify_worker::verify_torrent(Mediator& verify_mediator, bool const abort_flag)
 {
     verify_mediator.on_verify_started();
 
-    tr_sys_file_t fd = TR_BAD_SYS_FILE;
-    uint64_t file_pos = 0U;
-    uint32_t piece_pos = 0U;
-    tr_file_index_t file_index = 0U;
-    tr_file_index_t prev_file_index = ~file_index;
-    tr_piece_index_t piece = 0U;
-    auto buffer = std::vector<std::byte>(1024U * 256U);
-    auto sha = tr_sha1::create();
     auto last_slept_at = current_time_secs();
-
-    auto const& metainfo = verify_mediator.metainfo();
-    while (!abort_flag && piece < metainfo.piece_count())
+    for (tr_piece_index_t piece = 0U, n = verify_mediator.metainfo().piece_count(); !abort_flag && piece < n; ++piece)
     {
-        auto const file_length = metainfo.file_size(file_index);
+        verify_mediator.on_piece_checked(piece, verify_mediator.check_piece(piece));
 
-        /* if we're starting a new file... */
-        if (file_pos == 0 && fd == TR_BAD_SYS_FILE && file_index != prev_file_index)
+        /* sleeping even just a few msec per second goes a long
+         * way towards reducing IO load... */
+        if (auto const now = current_time_secs(); last_slept_at != now)
         {
-            auto const found = verify_mediator.find_file(file_index);
-            fd = !found ? TR_BAD_SYS_FILE : tr_sys_file_open(found->c_str(), TR_SYS_FILE_READ | TR_SYS_FILE_SEQUENTIAL, 0);
```

Comment:
```TR_SYS_FILE_SEQUENTIAL```
This is an important hint to OS.

Reply:
The flag will be passed to `tr_sys_file_open()` regardless of whether I revert the changes or not.

https://github.com/transmission/transmission/blob/76521a1751fb8b7638f524299943106c9c0917f2/libtransmission/open-files.cc#L194-L196